wagtail/users/views/users.py

```diff
@@ -180,57 +182,72 @@ def create(request):
     )
 
 
-@permission_required(change_user_perm)
-def edit(request, user_id):
-    user = get_object_or_404(User, pk=user_id)
-    can_delete = user_can_delete_user(request.user, user)
-    editing_self = request.user == user
+class Edit(EditView):
+
+    model = User
+    permission_policy = ModelPermissionPolicy(User)
+    form_class = get_user_edit_form()
+    template_name = "wagtailusers/users/edit.html"
+    index_url_name = "wagtailusers_users:index"
+    success_url = "wagtailusers_users:index"
+    edit_url_name = "wagtailusers_users:edit"
+    delete_url_name = "wagtailusers_users:delete"
+    success_message = _("User '{0}' updated.")
+    context_object_name = "user"
```

Comment:
The error message on submit is missing (when there are errors), here is what should be expected.

<img width="1000" alt="Screen Shot 2022-10-04 at 3 45 38 pm" src="https://user-images.githubusercontent.com/1396140/193743879-fbdccfee-cc3e-4ccb-aa87-22628f55e669.png">

```suggestion
    context_object_name = "user"
    error_message = gettext_lazy("The user could not be saved due to errors.")
```